DataLayer/Application/src/main/java/com/example/android/wearable/datalayer/MainActivity.kt

```diff
@@ -82,25 +88,42 @@ class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        val registry = WearDataLayerRegistry.fromContext(
+            application = this@MainActivity.applicationContext,
+            coroutineScope = lifecycleScope
+        )
+
+        dataClient = registry.dataClient
+        messageClient = registry.messageClient
+        capabilityClient = registry.capabilityClient
+
+        phoneDataLayerAppHelper = PhoneDataLayerAppHelper(
+            context = this,
+            registry = registry
+        )
+
         var count = 0
 
         lifecycleScope.launch {
-            lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED) {
-                // Set the initial trigger such that the first count will happen in one second.
-                var lastTriggerTime = Instant.now() - (countInterval - Duration.ofSeconds(1))
-                while (isActive) {
-                    // Figure out how much time we still have to wait until our next desired trigger
-                    // point. This could be less than the count interval if sending the count took
-                    // some time.
-                    delay(
-                        Duration.between(Instant.now(), lastTriggerTime + countInterval).toMillis()
-                    )
-                    // Update when we are triggering sending the count
-                    lastTriggerTime = Instant.now()
-                    sendCount(count)
+            if (phoneDataLayerAppHelper.isAvailable()) {
+                lifecycle.repeatOnLifecycle(Lifecycle.State.RESUMED) {
+                    // Set the initial trigger such that the first count will happen in one second.
+                    var lastTriggerTime = Instant.now() - (countInterval - Duration.ofSeconds(1))
+                    while (isActive) {
+                        // Figure out how much time we still have to wait until our next desired trigger
+                        // point. This could be less than the count interval if sending the count took
+                        // some time.
+                        delay(
+                            Duration.between(Instant.now(), lastTriggerTime + countInterval)
+                                .toMillis()
+                        )
+                        // Update when we are triggering sending the count
+                        lastTriggerTime = Instant.now()
+                        sendCount(count)
 
-                    // Increment the count to send next time
-                    count++
+                        // Increment the count to send next time
+                        count++
```

Comment:
I think this example is problematic. The count isn't stored or saved, so if there are reasons why the activity gets recreated it will revert to 0.

I think either

a) explain that the count is local to the specific instance of this activity and may reset
b) we use the DataClient as the current value and read and update the value.

The horologist sample might look more complicated with proto, and DataStore. But it's actually fundamentally a more real world example.